lib/cloudant.js

```diff
@@ -879,41 +879,54 @@ Cloudant.prototype.selectModel = function(model, migrate) {
 
 /**
  * Perform autoupdate for the given models. It basically calls db.index()
- *
+ * It does NOT destroy previous model instances if model exists, only
+ * `automigrate` does that.
  * @param {String[]} [models] A model name or an array of model names. If not
  * present, apply to all models
  * @param {Function} [cb] The callback function
  */
 Cloudant.prototype.autoupdate = function(models, cb) {
   debug('Cloudant.prototype.autoupdate %j', models);
-  this.automigrate(models, cb);
+  var self = this;
+  async.eachSeries(models, function(model, cb2) {
+    debug('Cloudant.prototype.autoupdate model %j', model);
+    var mo = self.selectModel(model, true);
+    self.updateIndex(mo, model, cb2);
```

Comment:
@b-admike I think here it should be `cb2`, it is the callback function of `async.eachSeries`'s iteration function.
The function next line is the "finish" function after all iterations done, in this function, we `cb(err)`, so err will be returned through `cb` if exist.